### PR TITLE
Introducing new dub-cli

### PR DESCRIPTION
**Pull request by sujjeee**
# `dub-cli`  #506
A CLI for easily shortening URLs

## Running Locally for Development
1. Clone the repository, install dependencies and navigate to the cli folder:
   ```bash
   cd packages/cli
   ```
2. Build the package in watch mode:
   ```bash
   pnpm dev
   ```
3. In a separate terminal, run commands:
   ```bash
   pnpm start [command]
   ```
4. See all available commands and options:
   ```bash
   pnpm start help
   ```
   
## Testing Production-like Setup

> **Warning**
> If you have previously installed `dub-cli` globally, uninstall it first to avoid conflicts

1. Build the package:
   ```bash
   pnpm build
   ```
2. Link the package globally:
   ```bash
    npm link
   ```
3. Verify the installation:
   ```bash
    dub -v
   ```
4. Run commands:
   ```bash
   dub [command]
   ```

### CODE REVIEW

File: packages/cli/src/commands/domain.ts [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+  .action(async () => {
+    const spinner = ora("Fetching domains").start();
+
+    try {
+      const showDomains = async () => {
+        const domains = await getDomains();
+        if (domains) spinner.stop();
+
+        return [
+          ...domains.map((domain) => ({
```
Comment (devkiran) [inline packages/cli/src/commands/domain.ts]: @sujjeee You can use `/api/domains/default` to fetch the default domains enabled for a workspace.

**Comment by devkiran** [comment]
@sujjeee Excited to see the DUB CLI. Let me know when it is ready for review.

We should limit the first version on CLI to shorting links, listing, etc. Domain management can now be done via the dashboard (It is just a one-time setup).

File: packages/cli/src/commands/domain.ts (hunk 1 above)
Comment (sujjeee) [reply packages/cli/src/commands/domain.ts]: should i remove domain related command?

Comment (devkiran) [reply packages/cli/src/commands/domain.ts]: Let's remove it and keep the first version as simple as possible.

**Comment by devkiran** [comment]
@sujjeee Can you fix the conflicts? Can you also share some notes on how to test this locally?

**Comment by sujjeee** [comment]
@devkiran anything else do i need to fix ?

**Comment by steven-tey** [comment]
@sujjeee can you resolve the merge conflicts here please?

**Comment by sujjeee** [comment]
@steven-tey i tired but didn't get any conflict on my end. 
image
any other way i can fix this ??

**Comment by sujjeee** [comment]
@steven-tey please update `getPackageInfo`  once it's published

**Pull request merged**